### PR TITLE
Change language selection window to support many languages

### LINKED ISSUE #6059: Main menu, language list is too big for native resolution
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.

### Describe the problem requiring a solution

Game supports too many languages now to fit the the screen at 640x480.
image

### Describe the possible solution

The list could be divided into to columns or completely reworked to have scroll bar.

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by ihhub**
This is definitely not a great solution due to code duplication but to reduce this it takes a lot of time.

close #6059

image

### CODE REVIEW

**Review by Branikolog — APPROVED** [review]
Hi, @ihhub !
Language selection works fine! 👍

File: src/fheroes2/dialog/dialog_language_selection.cpp [hunk 1]
```diff
@@ -36,97 +39,179 @@
...
+                                                                                   { 0, 0, originalSlider.width(), 8 }, { 0, 7, originalSlider.width(), 8 } );
+        items.setScrollBarArea( { roi.x + 328, roi.y + 73, 12, 180 } );
+        items.setScrollBarImage( scrollbarSlider );
+        items.SetAreaMaxItems( 11 );
+        items.SetAreaItems( { roi.x + 41, roi.y + 55 + 3, editBoxLength, 215 } );
+
+        std::vector<fheroes2::SupportedLanguage> temp = languages;
+        items.SetListContent( temp );
+
+        const fheroes2::Size currentResolution( display.width(), display.height() );
```
Comment (oleg-derevenetz) [inline src/fheroes2/dialog/dialog_language_selection.cpp:155]: What is the purpose of this variable?

File: src/fheroes2/dialog/dialog_language_selection.cpp [hunk 2]
```diff
@@ -36,97 +39,179 @@
...
+
+        fheroes2::Size selectedResolution;
```
Comment (oleg-derevenetz) [inline src/fheroes2/dialog/dialog_language_selection.cpp:157]: What is the purpose of this variable?

**Review by zenseii — APPROVED** [review]

**Review by oleg-derevenetz — CHANGES_REQUESTED** [review]
Hi @ihhub it looks like I found some leftovers :) Please take a look, maybe I missed something.

File: src/fheroes2/dialog/dialog_language_selection.cpp (hunk 1 above)
Comment (ihhub) [reply src/fheroes2/dialog/dialog_language_selection.cpp:155]: Copy-paste bug🙂

File: src/fheroes2/dialog/dialog_language_selection.cpp (hunk 2 above)
Comment (ihhub) [reply src/fheroes2/dialog/dialog_language_selection.cpp:157]: Copy-paste bug🙂

**Review by oleg-derevenetz — APPROVED** [review]

**Pull request merged**